.eslintrc

```
@@ -0,0 +1,58 @@
+{
+  "parser": "@typescript-eslint/parser",
+  "extends": [
+    "airbnb",
+    "plugin:@typescript-eslint/recommended",
+    "prettier" // prettier *has* to be the last one, to avoid conflicting rules
+  ],
+  "ignorePatterns": ["pnpm-lock.yaml"],
+  "plugins": ["simple-import-sort", "@typescript-eslint"],
+  "rules": {
+    "quotes": ["error", "double"],
+    "react/react-in-jsx-scope": "off", // next does not require react imports
+    "import/extensions": "off", // file extension not required when importing
+    "react/jsx-filename-extension": "off",
+    "no-restricted-syntax": "off",
+    "no-underscore-dangle": "off",
+    "no-await-in-loop": "off",
+    "react/jsx-props-no-spreading": "off",
+    "react/require-default-props": "off",
+    "simple-import-sort/imports": "warn",
+    "simple-import-sort/exports": "warn",
+    "import/first": "warn",
+    "import/newline-after-import": "warn",
+    "import/no-duplicates": "warn",
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-unused-vars": ["error"],
+    "@typescript-eslint/ban-types": "off",
+    "no-console": [
+      "error",
+      {
+        "allow": ["warn", "error", "debug"]
+      }
+    ],
+    "no-continue": "off",
+    "operator-linebreak": "off",
+    "max-len": "off",
+    "array-callback-return": "off",
+    "implicit-arrow-linebreak": "off",
+    "@typescript-eslint/no-non-null-asserted-optional-chain": "off",
+    "@typescript-eslint/no-non-null-assertion": "off",
+    "no-restricted-imports": "off",
+    "no-restricted-exports": "off",
+    "@typescript-eslint/ban-ts-comment": "off",
+    // TO FIX:
+    "import/no-cycle": "off", // pathpidia issue
+    "import/prefer-default-export": "off"
```

Those 2 rules are really great when working with requests and responses (it might not be a concern for app-sdk but it saved me multiple times in saleor-app-checkout 😅) - basically if you have a function that returns a promise you *must* await it or use `void` to explicitly mark the promise as ignored. You also cannot use promise in `if` statement + it covers other pitfalls

```ts
const myValue = getPromise(); // oh no I forgot to await
if (myValue) {} // this will always be true
```


```suggestion
    "import/prefer-default-export": "off",
    "@typescript-eslint/no-misused-promises": ["error"],
    "@typescript-eslint/no-floating-promises": ["error"]

```